hpx/runtime/components/client.hpp

```diff
@@ -0,0 +1,93 @@
+//  Copyright (c) 2007-2015 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(HPX_COMPONENTS_CLIENT_JUN_25_2015_0145PM)
+#define HPX_COMPONENTS_CLIENT_JUN_25_2015_0145PM
+
+#include <hpx/config.hpp>
+#include <hpx/runtime/components/client.hpp>
+#include <hpx/lcos/future.hpp>
+#include <hpx/util/move.hpp>
+
+namespace hpx { namespace components
+{
+    template <typename Component>
+    class client : public client_base<client<Component>, Component>
+    {
+        typedef client_base<client<Component>, Component> base_type;
+        typedef typename base_type::future_type future_type;
+
+    public:
+        // construction
+        client() {}
+
+        explicit client(naming::id_type const& gid)
+          : base_type(lcos::make_ready_future(gid))
+        {}
+        explicit client(naming::id_type && gid)
+          : base_type(lcos::make_ready_future(std::move(gid)))
+        {}
```

Comment:
would it make sense here to disable the `id_type` constructors taking the component id and instead create a forwarding constructor, which takes the locality id and the constructor arguments?